extensions/src/main/java/org/pentaho/platform/plugin/services/metadata/SessionCachingMetadataDomainRepository.java

```diff
@@ -397,23 +422,65 @@ public void removeModel( final String domainId, final String modelId ) throws Do
   @Override
   public Set<String> getDomainIds() {
     final IPentahoSession session = PentahoSessionHolder.getSession();
-
     final String domainKey = generateDomainIdCacheKeyForSession( session );
+    return getDomainIdsHelper( session, domainKey, () -> delegate.getDomainIds() );
+  }
+
+  @Override
+  public Set<String> getMetadataDomainIds() {
```

Comment:
[#1 Add function to get only Metadata Domains Ids and add functions to get only DataSourceWizards Domains Ids](https://jira.pentaho.com/browse/BACKLOG-36127)